apps/web/src/data-access/admin.ts

```diff
@@ -0,0 +1,206 @@
+import { eq, and } from 'drizzle-orm'
+import { database } from '@latitude-data/core/client'
+import { NotFoundError } from '@latitude-data/core/lib/errors'
+import { Result } from '@latitude-data/core/lib/Result'
+import {
+  workspaces,
+  users,
+  memberships,
+  projects,
+  features,
+  workspaceFeatures,
+} from '@latitude-data/core/schema'
+
+export type WorkspaceWithDetails = {
+  id: number
+  name: string
+  createdAt: Date
+  users: Array<{
+    id: string
+    email: string
+    name: string | null
+  }>
+  projects: Array<{
+    id: number
+    name: string
+  }>
+  features: Array<{
+    id: number
+    name: string
+    description: string | null
+    enabled: boolean
+  }>
+}
+
+export type UserWithDetails = {
+  id: string
+  email: string
+  name: string | null
+  workspaces: Array<{
+    id: number
+    name: string
+  }>
+}
+
+export type ProjectWithDetails = {
+  id: number
+  name: string
+  workspace: {
+    id: number
+    name: string
+  }
+}
+
+/**
+ * Find a workspace by ID for admin purposes (global access)
+ */
+export async function findWorkspaceByIdForAdmin(id: number, db = database) {
+  try {
```

Comment:
If someone use these data access without checking the user is admin it will reveal the data. 

The data layer is responsible for securing access. Pass the current user and do the check in all the methods